README.md

```diff
@@ -57,120 +47,95 @@ way yield from L1.
+- `rewAAVE` - ERC20 representing Aave reward token on L2.
 - `bridge` - bridge responsible for:
-  - minting and burning `static_a_tokens` on message from L1
-  - bridging `rewAAVE` tokens back to L1
-  - updating `rewards_index` for each `static_a_token` on message from L1
-- `proxy` - generic implementation of a proxy in cairo
+  - minting and burning `static_a_tokens` on message from L1.
+  - bridging `rewAAVE` tokens back to L1.
+  - updating `rewards_index` for each `static_a_token` on message from L1.
+- `proxy` - generic implementation of a proxy in cairo.
 
 ### More about static_a_token on L2
 
-Natively, Aave tokens grow in balance, not in value. To be able to create this
-kind of model, it is important to wrap them before bridging, converting them in
-a token that grows in value, not in balance.
+These `static_a_tokens` are a starting point for almost any cross-chain liquidity development to minimize “active” communication between chains. By design, a holder of those tokens - on Ethereum or after bridging somewhere else - will be passively accumulating yield from Aave on Ethereum.
 
-`static_a_tokens` are an implementation of the wrapped `aTokens` that will
-continuously increase in value on Starknet because they are backed by the
-increasing `aTokens` amounts locked in the bridge contract on Ethereum.
-`static_a_tokens` can then be bridged back to `aTokens`.
+More precisely, `static_a_tokens` are wrapped `aTokens` that grow in value while `aTokens` grow in balance. Such behavior is possible because `static_a_tokens` are back by increasing amounts of `aTokens` locked in the L1 counterpart of the bridge. `static_a_tokens` living on L2 can be bridged back to `aTokens`.
 
 ### Proxies
 
-All calls made to the following contracts will be handled by a proxy who
-delegates the calls to the available implementation of these contracts.
+Each of the following contracts is deployed behind a proxy:
 
 - `bridge` on L2
-- `static_a_token`s on L2
-- `Bridge` on L1
+- `static_a_token` on L2
 - `rewAAVE` token on L2
+- `Bridge` on L1
 
 ### Governance
 
-- `static_a_token`s are controlled by L2 `bridge`.
+- `static_a_token` deployed contracts are controlled by L2 `bridge`.
 - `rewAAVE` token is owned by L2 `bridge`.
 
-## How does it work?
+## How it works
 
 ### Bridging aTokens from L1 to L2
 
+<a name="approve-bridge-tokens"></a>
 **Approve bridge tokens**
 
-L1 aTokens are approved on the bridge at `initiliaze` where `_approveBridgeTokens` is called internally to approve the provided array of aTokens in an array along with their corresponding static_a_tokens on L2. :warning: Gas limit concerns should apply here!
+Pairs of (L1 aToken, L2 static_a_token) are approved on the bridge in `initiliaze` function. The function `_approveBridgeTokens` is called internally to approve an array of aTokens with their corresponding static_a_tokens on L2.
```

Comment:
Why not simply write: L1 aTokens with their corresponding static_a_tokens on L2, are going to be approved on the l1 bridge in  `initialize` function.